### PR TITLE
Core module updated

### PR DESCRIPTION
**Pull request by obedjamir**
Database entity classes created
Repository interfaces created

### CODE REVIEW

File: core/src/main/java/com/bankbazaar/tripmanager/model/Trip.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+@NoArgsConstructor
+@Table(name="trip")
+public class Trip {
+    @Id
+    @Column(name="trip_id")
+    @NotNull(message = "trip_id cannot be null")
+    private Long tripId;
+
+    @Column(name="trip_name")
+    @NotBlank(message = "tripName cannot be blank")
```
Comment (PinakiGhosh) [inline core/src/main/java/com/bankbazaar/tripmanager/model/Trip.java]: Dont use @NotBlank here. Column has an attribute nullable set that.

File: core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    private Long tripId;
+
+    @Column(name="activity_description",nullable = false)
+    private String activityDescription;
+
+    @Column(name="location",nullable = false)
+    private String location;
+
+    @Column(name="activity_time",nullable = false)
+    private Time activityTime;
```
Comment (PinakiGhosh) [inline core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java]: Make this Date

File: core/src/main/java/com/bankbazaar/tripmanager/model/TripUserMapping.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    @Id
+    @Column(name="trip_id",nullable = false)
+    private Long tripId;
+
+    @Id
+    @Column(name="user_id",nullable = false)
+    private Long userId;
+
+    @Column(name="user_role",nullable = false)
+    private String userRole;
```
Comment (PinakiGhosh) [inline core/src/main/java/com/bankbazaar/tripmanager/model/TripUserMapping.java]: Make this an enum

**Review by PinakiGhosh — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/bankbazaar/tripmanager/model/Trip.java (hunk 1 above)
Comment (obedjamir) [reply core/src/main/java/com/bankbazaar/tripmanager/model/Trip.java]: This has been resolved

File: core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java [hunk 4]
```diff
@@ -30,7 +30,7 @@ public class TripActivity implements Serializable {
     private String location;
 
     @Column(name="activity_time",nullable = false)
-    private Time activityTime;
+    private Data activityTime;
```
Comment (PinakiGhosh) [inline core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java]: Date?

**Review by PinakiGhosh — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java (hunk 4 above)
Comment (obedjamir) [reply core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java]: Oh my bad. missed that one

**Pull request merged**